ChatApp/Scenes/Chat/ChatAppViewController.swift

```diff
@@ -0,0 +1,217 @@
+                self?.bottomChatView.setUpUsers(sender: secondUser, recipient: firstUser)
+            }
+        })
+    }
+    
+    //MARK: - Status Bar  Style
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return statusBar
+    }
+    
+    //MARK: - KeyBoard
+    private func keyboardDismiss() {
+        let tapGesture = UITapGestureRecognizer(target: view, action: #selector(UIView.endEditing))
+        view.addGestureRecognizer(tapGesture)
+    }
+    
+    //MARK: - Chat View Methods
+    private func setUpTopChatView() {
+        view.addSubview(topChatView)
+        topChatView.delegate = self
+        
+        NSLayoutConstraint.activate([
+            topChatView.topAnchor.constraint(equalTo: switcherView.bottomAnchor),
+            topChatView.bottomAnchor.constraint(equalTo: dividerView.topAnchor ),
+            topChatView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            topChatView.leadingAnchor.constraint(equalTo: view.leadingAnchor)
+        ])
+    }
+    
+    private func setUpBottomChatView() {
+        view.addSubview(bottomChatView)
+        bottomChatView.delegate = self
+        
+        NSLayoutConstraint.activate([
+            bottomChatView.topAnchor.constraint(equalTo: dividerView.bottomAnchor),
+            bottomChatView.bottomAnchor.constraint(
+                equalTo: view.bottomAnchor,
+                constant: Constants.ChatView.bottomChatViewPadding
+            ),
+            bottomChatView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            bottomChatView.leadingAnchor.constraint(equalTo: view.leadingAnchor)
+        ])
+    }
+    
+    //MARK: - Divider View
+    private func setUpDividerView() {
+        view.addSubview(dividerView)
+        dividerView.setHeight(Constants.DividerView.dividerViewHeight)
+        dividerView.backgroundColor = Constants.Color.dividerViewYellowBackgroundColor
+        
+        NSLayoutConstraint.activate([
+            dividerView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            dividerView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            dividerView.leadingAnchor.constraint(equalTo: view.leadingAnchor)
+        ])
+    }
+    
+    //MARK: - Switcher View
+    private func setUpSwitcherView() {
+        view.addSubview(switcherView)
+        switcherView.delegate = self
+        
+        NSLayoutConstraint.activate([
+            switcherView.topAnchor.constraint(
+                equalTo: view.topAnchor,
+                constant: Constants.SwitcherView.switcherTopAnchor
+            ),
+            switcherView.trailingAnchor.constraint(
+                equalTo: view.trailingAnchor,
+                constant: Constants.SwitcherView.switcherRightAnchor
+            )
+        ])
+        
+        let darkModeEnabled = UserDefaults.standard.bool(forKey: Constants.UserDefaults.key)
```

Comment:
მე გირჩევდი რომ UserDefaults Manager გაგეკეთებინე და სტატიკური პარამეტრით 
`public static var isDarkMode: Bool {
        get {
            return UserDefaults.standard.bool(forKey: Constants.UserDefaults.key) // It returns false if not found
        } set {
            UserDefaults.standard.set(newValue, forKey: Constants.UserDefaults.key)
        }
    }`